### PR TITLE
docs: Join on prem clusters to fleet hub cluster

### PR DESCRIPTION
**Pull request by Arvindthiru**
### Description of your changes

Fixes #

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: docs/tutorials/JoinOnPremClustersToFleet.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+Clone the [fleet-networking repository](https://github.com/Azure/fleet-networking) and navigate to the root directory of the repo.
+
+Run the following script to install the fleet networking member agents on each on-prem cluster.
+
+The latest fleet-networking image tag could be found here [fleet-networking releases](https://github.com/Azure/fleet-networking/releases).
+
+Ex: `./hack/Azure/joinMC.sh v0.1.0 v0.2.0 hub test-cluster-1 test-cluster-2`
+
+```shell
+./hack/membership/joinMC.sh <FLEET-IMAGE-TAG> <FLEET-NETWORKING-IMAGE-TAG> <HUB-CLUSTER-NAME> <MEMBER-CLUSTER-NAME-1> <MEMBER-CLUSTER-NAME-2> <MEMBER-CLUSTER-NAME-3> ...
```
Comment (britaniar) [inline docs/tutorials/JoinOnPremClustersToFleet.md:57]: include chmod + x  ./hack/membership/joinMC.sh

File: docs/tutorials/JoinOnPremClustersToFleet.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+> **Note:** The script in the fleet-networking repo should only be run once the script in the fleet repo has been run to ensure the member agents can communicate with the hub cluster.
+
+Clone the [fleet-networking repository](https://github.com/Azure/fleet-networking) and navigate to the root directory of the repo.
+
+Run the following script to install the fleet networking member agents on each on-prem cluster.
+
+The latest fleet-networking image tag could be found here [fleet-networking releases](https://github.com/Azure/fleet-networking/releases).
+
+Ex: `./hack/Azure/joinMC.sh v0.1.0 v0.2.0 hub test-cluster-1 test-cluster-2`
```
Comment (britaniar) [inline docs/tutorials/JoinOnPremClustersToFleet.md:54]:
```Suggestion
Ex: `./hack/membership/joinMC.sh v0.1.0 v0.2.0 hub test-cluster-1 test-cluster-2`
```

**Comment by Arvindthiru** [comment]
Reverted changes to hack/Azure/setup directory and moved the script to hack/membership to be consistent with fleet-networking repo

File: hack/membership/cleanup.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+fi
+
+for MEMBER_CLUSTER in "${@:2}"; do
+if [[ ! $(kubectl config view -o jsonpath="{.contexts[?(@.context.cluster==\"$MEMBER_CLUSTER\")]}") ]] > /dev/null 2>&1; then
+  echo "The cluster named $MEMBER_CLUSTER does not exist."
+  exit 1
+fi
+done
+
+export CONNECT_TO_FLEET=connect-to-fleet
```
Comment (britaniar) [inline hack/membership/cleanup.sh:22]: Is this being used?

File: hack/membership/cleanup.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+export HUB_CLUSTER="$1"
```
Comment (britaniar) [inline hack/membership/cleanup.sh:24]: defined twice (line 9)

File: hack/membership/cleanup.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+export MEMBER_CLUSTER_CONTEXT=$(kubectl config view -o jsonpath="{.contexts[?(@.context.cluster==\"$MEMBER_CLUSTER\")].name}")
+
+kubectl config use-context $HUB_CLUSTER_CONTEXT
+kubectl delete secret $MEMBER_CLUSTER-hub-cluster-access-token -n connect-to-fleet
+kubectl delete serviceaccount $MEMBER_CLUSTER-hub-cluster-access -n connect-to-fleet
+kubectl config use-context $MEMBER_CLUSTER_CONTEXT
+helm uninstall member-agent
+helm uninstall member-net-controller-manager
+helm uninstall mcs-controller-manager
+done
```
Comment (britaniar) [inline hack/membership/cleanup.sh:38]: Does the user have to delete the namespace? I assume after all memberclusters are cleaned up?

File: docs/tutorials/JoinOnPremClustersToFleet.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
+# Tutorial: Join On-Prem Clusters to Fleet hub cluster
```
Comment (ryanzhang-oss) [inline docs/tutorials/JoinOnPremClustersToFleet.md:1]: This is Azure specific, should be under hack/Azure

File: docs/tutorials/Azure/JoinOnPremClustersToFleet.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+# Steps to make an on-prem cluster leave the Fleet hub cluster
+
+Delete the `MemberCluster` resource for a particular on-prem cluster in the hub cluster.
+
+The join script in the fleet repo creates `MemberCluster` resource with the same name as your on-prem cluster.
+Replace <cluster-name> with the name of your on-prem cluster.
+
+```
+kubectl config use-context hub
+kubectl delete membercluster <cluster-name>
```
Comment (ryanzhang-oss) [inline docs/tutorials/Azure/JoinOnPremClustersToFleet.md:99]:
```suggestion
kubectl delete membercluster <cluster-name> --wait
```

**Review by ryanzhang-oss — APPROVED** [review]

**Review by britaniar — APPROVED** [review]

**Pull request merged**